source/code/plugins/out_oms.rb

```diff
@@ -95,6 +95,15 @@ def write(chunk)
       unmergable_records.each { |tag, record|
         handle_record(tag, record)
       }
+    rescue OMS::RetryRequestException => e
+      @log.debug "Encountered retryable exception. Will retry sending data later. #{e}"
+      # Re-raise the exception to inform the fluentd engine we want to retry sending this chunk of data later.
```

Comment:
It's already handled by the engine and is configurable. http://docs.fluentd.org/articles/output-plugin-overview#retrywait-maxretrywait